### PR TITLE
Delay shutdown when remove or undeploy NS

### PR DESCRIPTION
**Pull request by Mykhailenko**

### CODE REVIEW

**Comment by fviale** [comment]
after looking at the code, it seems quite difficult to double check shutdown has been called inside a test

File: rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java [hunk 1]
```diff
@@ -2567,7 +2603,8 @@ private void validateNodeSourceNameOrFail(String nodeSourceName) {
         }
         if (this.definedNodeSources.containsKey(nodeSourceName) ||
             this.deployedNodeSources.containsKey(nodeSourceName) ||
-            this.delayedNodeSourceRemovalEvents.containsKey(nodeSourceName)) {
+            this.delayedNodeSourceRemovalEvents.containsKey(nodeSourceName) ||
```
Comment (fviale) [inline rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: this check is a bit strange, is it meant to handle the following scenario ?

1) user removes a ns, waiting on busy nodes
2) It takes too long, user try again to remove the ns.

Comment (Mykhailenko) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: @fviale did not get your point. This check here is to ensure we dont create new NS with a name that is already exist

Comment (fviale) [reply rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java]: sorry, code folding misunderstanding

**Pull request merged**